### DISCUSSION TITLE
2023-04-14 - The best way to handle unexpected requests mid-Sprint - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by lukecookssw** [SSWConsulting/SSW.Rules.Content#5361] [Chewing the Fat]
Hi team,

Let's discuss this rule: [Do you have flexibility with the order you do the work?](https://ssw.com.au/rules/unexpected-requests)

**Comment by tiagov8** [comment]
Communication is key - As long as everyone is in sync and aware of the effects of a Sprint change I don't see a lot of harm in it.

**Comment by jerryluossw** [comment]
Great rule, in the long term, maintaining the team's work pace is more important than asking to work overtime to deal with temporary requests.

In China, especially for government clients, they often add requests mid sprints, the way we deal with it is as similar as the rule, most of time it works, and clients like the way of communication very much, that's one of the reasons why they think SSW is better than other suppliers. But sometimes they don't want to delete other PBIs （Usually because their boss wants everything）, so we have to do some overtime work, luckly those cases aren't often.

**Comment by leehawkinsssw** [comment]
The rule is helpful and serves as a handy reference for anyone encountering this issue during their daily work. 

It's a common problem with stakeholders who are new to Scrum and dev teams are often their own worst enemies in taking on such requests to try to keep stakeholders happy. 

The Sprint is an important (I wouldn't use the term "sacred") container for work that easily overflows if more stuff is added without draining some other stuff out. One of the benefits of short Sprints is the easier conversations you can have about these "urgent" additional pieces of work - "can this really not wait for 2 weeks and be added to the next sprint?"

**Comment by leahy268** [comment]
I haven't always done Scrum.  However I would say that it's always important to be prepared to push back on a Product Owner.  I'm not saying don't do the work but you must prioritize it in amongst all the other tasks that you already have.  

It's amazing how often when a Product Owner realizes that something else has to be pushed back to fit in a 'Urgent' task how often the 'Urgent' task becomes only an 'Important' task.

**Comment by jakebayliss** [comment]
Super important rule when it comes to keeping the sprint goal alive. I think if the new task is big enough to cause the team to likely miss the end goal then put it in the backlog, but if its only small then adding it to the sprint is probably fine

**Comment by bahjatalaadelssw** [comment]
This is a useful skill because it helps manage over committing to tasks. Without set expectations it's easy to burnout or start to despise your work because the goal post keep getting moved. Staying balanced is key.

**Comment by ozdemir-mehmet** [comment]
Agree with all of the above

**Comment by danielmackay** [comment]
Great video @jontygardner!  Nice to see you in front of the camera for a change. 🙂

Some great tips here for avoiding distraction and dealing with stakeholders/PO

**Comment by vladislav-kir** [comment]
Great tips for keeping Sprint on track when facing interruptions.

**Comment by JackDevAU** [comment]
Great video + rule - with a good email template! Nice work! 

Communication is a must and this rule will defiantly help when dealing with ad-hoc work!

**Comment by calumjs** [comment]
Protect your Sprints, be ready for curveballs, and keep the team happy and productive. That's the key to rockin' Scrum!

**Comment by TobyChurches** [comment]
Great Video and a Great Rule! Definitely important to maintain the sprint workload and ensure that you arent over committing to tasks.

**Comment by cvionnet** [comment]
Nice video and rule!
Maintaining the focus of the Sprint is super important.
Additionally, it's important to educate stakeholders who may not be familiar with Scrum about its benefits.

**Comment by jontygardner** [comment]
I enjoyed making this video as I have found this approach to Scrum very effective. 

Even if you receive 'small tasks' from the PO, make sure you follow this method and create a PBI for it. Document the work so you have a trail. The exception for this is if the work takes than 30 minutes to complete. But then again, if the Product Owner gives you 10 x 30 minute Ad Hoc tasks then that compromises the Sprint Goals.

**Comment by Ben0189** [comment]
Nice Video, the request's urgency is important things to consider before made any adjustment on the sprint plan

**Comment by tombui99** [comment]
This is a very useful rule! 
In my own experience, my client has always been adding new task mid Sprint, however, communication is always the key to a successful project as long as you are in sync with the client and they acknowledge that it will take more time to finish the Sprint.

**Comment by AntPolkanov** [comment]
Scope creep can be a real pain and may heart the team. It's a gift to have a Scrum Master on your team who is ready to negotiate and even push back.

**Comment by sethdaily** [comment]
As someone who new to Scrum, the rule and video are a good introduction to the importance of Sprints. It is a clear explanation of how the Sprint framework ensures that a team doesn't overcommit themselves, while also keeping flexibility in the development process.

**Comment by amankumarrr** [comment]
I agree, regular ad-hoc work can have a significant impact on the team's progress.

**Comment by Hona** [comment]
I think the Developers are just as much responsible for the backlog & sprint. (Manage up rule)

If you spot a problem that could occur from new work being allocated, make sure to mention it, and get it dealt with quickly.

**Comment by Abf09** [comment]
It can have a significant impact. However, if the client understands the outcome of regular ad-hoc work and cooperates on time consumption, it shouldn't affect the progress.

**Comment by AlexRothwell** [comment]
Having a strong Scrum Master on the team makes handling these situations far easier. In my recent experience, they can help reduce interruptions to Sprints in progress and manage expectations from the PO

However, it's everyone's responsibility to maintain the Scrum practices, whether that's just following the usual ceremonies or helping to coach a PO who is new to Scrum

**Comment by chrisschultzssw** [comment]
This is useful outside of Scrum as well - e.g. in a ticketing system, you can tell the PO "I'm currently working on ticket #XXXX, if this request is more important I can stop working on it, but we won't meet our SLA".

**Comment by JackDuan123** [comment]
I will not be too anoyed if my client phone me and say "Jack I want to ask you to help me do some urgent task on..." becuase I understand my clients have their clients and sometimes my job is to help them satisfy their client.
In a customer central culture your customer 's customer is  the god of your god, or your Gnostic god. I don't like Gnostic but if there is really one exists, I would worship it.
And what I firmly believe is always go with the bigger rule, we did customer service first and software development second, so I think we should follow the customer service rules more than developer rules

**Reply by steven0x51** [reply]
"Jack I want to ask you to help me do some urgent task on..." That sounds like spam 🤣

**Reply by JackDuan123** [reply]
I have this kind of spam occasionally :) :)

**Reply by leehawkinsssw** [reply]
Are you saying you disagree with the rule, Jack? If so, how could it be improved?

**Reply by JackDuan123** [reply]
I honestly don't go against the rule but I just feel like we should always keep our client comfortable, nobody really likes to be educated.

**Comment by tomek-i** [comment]
Great Video and Rule. 

I've seen some good comments here. I also think it is important to educate the Client / Stakeholder.
From previous work experience I realised that **pushing back** part is very Important and that the Scrum Team should not be _afraid_ to do so. If they give in eg. the Devs accept incomplete PBI's into the sprint or the Product Owner allows scope creep or makes constant changes mid sprint because the client/stakeholder wants to, then it adds risk to the sprint and frustration builds up.

**Comment by JeanThirion** [comment]
In the real life, the Scrum Master has the unwritten responsibility of deciding whether or not ad-hoc work is beneficial to the client or not. That's the real hard bit of Scrum Master in consultancy Dev Teams. You have to deeply understand what the client's business is all about, and help the PO define "real" priorities.

**Comment by steven0x51** [comment]
Great video! I do agree to double-check the importance of the new item, if it was less important then it is not worth changing the plan in the middle of the sprint.

**Comment by zacharykeeping** [comment]
Good video, and good rule for keeping a sprint focused and everybody happy with good communication.

**Comment by jernejk** [comment]
The video describes really well how to deal with ad-hoc work.

The only thing I would add is that we can add an option of adding more people to handle more requests, especially if the people added already worked on the project and it's cheap to add them.


Potentially a bit off-topic, the issue I have the most on ad-hoc work is SSW work during client work. It's hard to determine when to do it and it accumulates as time goes by. I'm trying to offload the SSW email tasks to Trello and hopefully, that gives me a better system to deal with this problem as most tasks are unmanaged emails.

**Comment by matt-goldman** [comment]
Great rule and vide!

**Comment by christoment** [comment]
I think pushing ad-hoc work is always a good strategy. It also helps the PO to have a second pass to think about whether the task is urgent or not :)

**Comment by tkapa** [comment]
Great rule :)

**Comment by AttackOnMorty** [comment]
While this rule is specific to Scrum, its methodology can be applied to other work modes as well. Check the priority of the new task, then update the task list and notify others.

**Comment by stephanfakossw** [comment]
Great video and message. @jontygardner We will be on the lookout for more videos from you (Infront instead of behind the camera).

**Comment by AshAnil** [comment]
The rule provides ample guidance on fostering a cordial and seamless relationship between the team members and the Product Owner, making Scrum an effective methodology for exhibiting our ongoing tasks and ensuring transparency that pleases the Product Owner.

**Comment by Harry-Ross** [comment]
I thought this was a great video, with some really important points about how Scrum helps out teams. I also really loved the graphics - made the video really engaging.

**Comment by landonmaxwell** [comment]
Removing unfinishable tasks when new ones are added is good for morale, as team members won't feel like they have "failed" a sprint by having a heap of work left that was unrealistic to complete due to interruptions.

**Comment by tino-liu** [comment]
This is a great video!
It really is very important that client and developers have a mutual understanding, and SCRUM is the perfect bridge to build that understanding. I've had experience with my previous company where I get dumped a few bugs while working on a user story. Sometimes working these extra hours trying to "fix" everything ends up being counterproductive.

**Comment by jimmy1992abc** [comment]
The video is great, I am happy that getting to watch more structured videos like this. Impressive and informative.

Generally I believe pushing ad-hoc tasks to the current sprint is costy both in time and productivity. We need to avoid compromising stakeholders at the cost of low productivity and communicate with the stakeholders properly to make sure they understand it. I prefer to include the tasks that would be more important than any of the current PBIs.
(Powered by Jimmy Chen)

Ad-hoc tasks can be challenging to handle, especially if they disrupt our plans and schedules. However, it's important to keep in mind that sometimes these tasks are genuinely important and need to be addressed quickly. In such cases, it may be necessary to rearrange our plans and schedules to accommodate them.

It's worth noting that effective time management involves prioritizing tasks based on their importance and urgency. If we're able to identify which tasks are most critical and plan accordingly, we may be better equipped to handle unexpected ad-hoc tasks when they arise.

Ultimately, it's important to communicate with the product owner or whoever is assigning the task to understand its urgency and importance. By working together and being transparent about our capacity and availability, we can strive to find a solution that works for everyone.
(Powered by ChatGPT not 4😥)

**Comment by JaneYangssw** [comment]
Nice video and great rule!  It's so important to consider the team's interests before making any changes in a sprint and this rule really helps us to do that. It's great to see such a thoughtful and effective approach to team collaboration.

**Comment by Aibono1225** [comment]
Great video, short and informative. Since unexpected requests happen occasionally, we really should build bridges between developers and POs to meet the requirements of both parties as much as possible

**Comment by jimmidier** [comment]
Great video and rule. It's also important to be mindful when breaking down PBI into tasks. The smaller the tasks the better the granularity will be, thus more prepared for any change/cancellation mid-Sprint

**Comment by drwharris** [comment]
Interruptions are inevitable. One of the goals of scrum is to allow you to pivot quickly. If a new, higher priority issue comes up then you just have to pivot. However its important that the decision on why you pivoted is recorded. 

1. Make sure a PBI is logged for the pivot work. 
2. Have the decision maker record the need for the pivot in your current PBI & the new PBI.

**Comment by sylhuang** [comment]
The beauty of Scrum is we are more flexible and reactive to new changes. If absolutely necessary, we are able to take out PBIs of similar size and swap for a new prioritized work. Like the game of packing a box.

**Comment by KristenHu** [comment]
Great video, there are always emergency situations in real life, which is hard to avoid, this rule helps a lot to tell us how to communicate with the PO and the client to solve these problems.

**Comment by AnkyChe** [comment]
Great video and great tips!
Interruptions are inevitable so the conmmunication is important. The vedio teaches me how to do that communication -  first acknowledge it (e.g. "That's a great idea!"), and then enquire about the new task's priority.

**Comment by SofieHong** [comment]
Great video and rule! 👍 Especially the PBI Substitutions email template, it provides an effective way to deal with PBI changes.

**Comment by Berber-Mama** [comment]
I love the video. Protecting Sprint's progress is as important as balancing clients' needs!

**Comment by PothieuG** [comment]
Great video and content.
Nice tip as well, I'll try to use it well in my future project !

**Comment by SamBWagner** [comment]
Just be clear with the PO that the Sprint Goal is the 1 and only real metric for measuring the team's success. If this isn't communicated to them early and clearly. It's best to pursue the conversation with them

**Comment by BrookJeynes** [comment]
Very important rule and serves as a good reference

**Comment by UlyssesMaclaren** [comment]
It's great that everyone's getting aligned on this :)

**Comment by camillars** [comment]
Great video from Jonty. These are great tips to keep the Sprint work/goal on track amidst interruptions, which, let's face it, will always come.

**Comment by Levijj22** [comment]
Great video @jontygardner!
Awesome to see everyone on the same page.
